src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -199,7 +229,21 @@ public void Refresh(string key)
             throw new ArgumentNullException(nameof(key));
         }
 
-        GetAndRefresh(key, getData: false);
+        var expandedKey = _expandKey(key);
+
+        Connect();
+
+        var value = _cache.HashGet(expandedKey, _refreshLabels);
```

Comment:
By doing the get separately first, we've doubled latency; I would have anticipated a much more literal translation of the original script, with the "get" inside the tran/batch, then everything is one-of latency (just like the script). Happy to offer examples in the new year.